### PR TITLE
[RAMP] Allow shuffle when auto-batching disabled in DataLoader

### LINKED ISSUE #35761: Allow shuffle when auto-batching disabled in `DataLoader`
## 🚀 Feature
Allow `DataLoader(batch_size=None, shuffle=True)`

## Motivation

My model handles one sample a time. So, I don't need auto-batching, but I want my dataset to be shuffled. However, when I use `DataLoader(batch_size=None, shuffle=True)`, it raises `batch_size=None option disables auto-batching and is mutually exclusive with shuffle, and drop_last`.

But I can't see why they are mutually exclusive.

## Pitch

Allow `DataLoader(batch_size=None, shuffle=True)`.

Basically, just remove the `shuffle` check from this line:  https://github.com/pytorch/pytorch/blob/ada647214f3c39b3fdf624dd62f8295c92144617/torch/utils/data/dataloader.py#L202

## Alternatives

I can:
* Use `batch_size=1` then remove the batching dimension afterwards
* Use `sampler=RandomSampler(dataset)`

But these makes me feel like I'm fighting with DataLoader. Why I cannot just use `shuffle=True`?

## Additional context

cc @SsnL

### PR DESCRIPTION
**Pull request by ydaiming**
Fix https://github.com/pytorch/pytorch/issues/35761
cc @SsnL

### CODE REVIEW

**Comment by choidongyeon** [comment]
It looks like these lines should be removed: https://github.com/pytorch/pytorch/blob/master/test/test_dataloader.py#L944-L950

**Comment by SsnL** [comment]
@choidongyeon is correct. Please remove that test. Also, could you add the test that this combination works here? https://github.com/pytorch/pytorch/blob/baf6ed0238df7a8f30655df5b3b265e77787e422/test/test_dataloader.py#L1224-L1225

You probably also need to fix `_test_shuffle` to work with `batch_size=None` case.
